CompilerSupport.cmake

```diff
@@ -0,0 +1,26 @@
+include(CheckCXXSourceCompiles)
+
+if(MSVC AND (MSVC_TOOLSET_VERSION GREATER_EQUAL 143))
+set (COMPILER_HAS_RANGES TRUE)
+elseif((CMAKE_CXX_COMPILER_ID EQUALS "GNU") AND (CMAKE_CXX_COMPILER_VERSION LESS_EQUAL 11))
+set (COMPILER_HAS_RANGES FALSE)
+else()
+check_cxx_source_compiles("
+#include <ranges>
+#include <vector>
+#include <algorithm>
+#include <string>
+
+int main() {
+    auto parts_view = std::string(\"Hello World\")
+        | std::views::split(std::string_view{\" \"}) // Split by space
+        | std::views::transform([](auto&& subrange) {
+            return std::string(subrange.begin(), subrange.end());
+        });
+
+    std::vector<std::string> processed_words;
+    std::ranges::copy(parts_view, std::back_inserter(processed_words));
+    return (2 == processed_words.size()) ? 0 : 1;
+}
+" COMPILER_HAS_RANGES)
+endif()
```

Comment:
Why not just do the compile test in all cases, that way if a future msvc compiler somehow breaks range support we catch it here.  Doing this compile check is pretty damned snappy and just doing it always feels better than those explicit checks.

Reply:
it isn't *that* snappy, and chances of a future compiler version breaking something that works are very low. You have to be able to trust some fundamental stuff.

Reply:
I suppose, I just tend to have an aversion to testing things with magic numbers (ie, 143 and 11). [and yes, I know the code is *rife* with those, and I want to get rid of them all eventually]